app/population/middleware/set-population-id.js

```diff
@@ -0,0 +1,24 @@
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+
+async function setResultsPopulationTable(req, res, next) {
+  try {
+    const { locationId, date } = req.params
+
+    const freeSpaces = await locationsFreeSpacesService.getPrisonFreeSpaces({
+      dateFrom: date,
+      dateTo: date,
+      locationIds: locationId,
+    })
+
+    const population = freeSpaces?.[0]?.meta?.populations?.[0] || {}
+    const { id: populationId } = population
```

Comment:
Could this be wrapped in a check so that `req.populationId` isn't set if there is no poplulation?

Reply:
Although, looking at the test it will still be set as undefined I guess so maybe it doesn't need it.

Reply:
Resolving as this has been discussed during pairing and been changed into a "setPopulation" param middleware